File: medlink/medlink.py
import os

def get_database_path():
    return os.path.join(os.path.dirname(__file__), 'MedicalDatabase.txt')

def add_record():
    name = input("Patient's Name: ").strip()
    address = input("Patient's Address: ").strip()
    phone = input("Patient's Phone: ").strip()
    email = input("Patient's Email: ").strip()
    note = input("Patient's Condition: ").strip()

    with open(get_database_path(), 'a') as file:
        file.write(f"{name}\n{address}\n{phone}\n{email}\n{note}\n--------------\n")
    
    print("Patient information added successfully to medical database.")

def delete_record():
    search_name = input("Enter Patient's name to delete record: ").strip()
    lines = []
    with open(get_database_path(), 'r') as file:
        lines = file.readlines()
    
    with open(get_database_path(), 'w') as file:
        skip = False
        for line in lines:
            if line.strip() == search_name:
                skip = True
            if not skip:
                file.write(line)
            if skip and line.strip() == '--------------':
                skip = False
    
    print("Record deleted successfully.")

def search_record():
    search_name = input("Enter Patient's name to search record: ").strip()
    found = False
    with open(get_database_path(), 'r') as file:
        for line in file:
            if line.strip().lower() == search_name.lower():
                found = True
                print(line, end='')
                continue
            if found:
                print(line, end='')
                if line.strip() == '--------------':
                    found = False
                    break
    
    print("Search completed.")

def edit_record():
    search_name = input("Enter Patient's name to edit record: ").strip()
    lines = []
    with open(get_database_path(), 'r') as file:
        lines = file.readlines()
    
    with open(get_database_path(), 'w') as file:
        skip = False
        for line in lines:
            if line.strip() == search_name:
                skip = True
            if not skip:
                file.write(line)
            if skip and line.strip() == '--------------':
                skip = False
    
    add_record()
    print("Record edited successfully.")

def main():
    while True:
        print("1. Add Patient Record")
        print("2. Delete Patient Record")
        print("3. Search Patient Record")
        print("4. Edit Patient Record")
        print("5. Exit Medlink")
        choice = input("Enter your choice: ").strip()

        if choice == '1':
            add_record()
        elif choice == '2':
            delete_record()
        elif choice == '3':
            search_record()
        elif choice == '4':
            edit_record()
        elif choice == '5':
            break
        else:
            print("Invalid choice")

if __name__ == "__main__":
    main()

File: medlink/medlink.pl
#!/usr/bin/perl
use strict;
use warnings;

sub add_record {
    print "Patient's Name: ";
    chomp(my $name = <STDIN>);
    print "Patient's Address: ";
    chomp(my $address = <STDIN>);
    print "Patient's Phone: ";
    chomp(my $phone = <STDIN>);
    print "Patient's Email: ";
    chomp(my $email = <STDIN>);
    print "Patient's Condition: ";
    chomp(my $note = <STDIN>);

    open(my $fh, '>>', 'MedicalDatabase.txt') or die "Could not open file 'MedicalDatabase.txt' $!";
    print $fh "$name\n$address\n$phone\n$email\n$note\n--------------\n";
    close($fh);
    print "Patient information added successfully to medical database.\n";
}

sub delete_record {
    print "Enter Patient's name to delete record: ";
    chomp(my $search_name = <STDIN>);

    # Read the entire file into an array
    open(my $fh, '<', 'MedicalDatabase.txt') or die "Could not open file 'MedicalDatabase.txt' $!";
    my @lines = <$fh>;
    close($fh);

    # Write back only the records that do not match the search name
    open($fh, '>', 'MedicalDatabase.txt') or die "Could not open file 'MedicalDatabase.txt' $!";
    my $skip = 0;
    foreach my $line (@lines) {
        if ($line =~ /^$search_name$/) {
            $skip = 1; # Start skipping the lines for the current record
        } elsif ($line =~ /^--------------$/) {
            $skip = 0; # Stop skipping at the end of the record
        } elsif (!$skip) {
            print $fh $line; # Write lines that are not part of the record to be deleted
        }
    }
    close($fh);

    print "Record deleted successfully.\n";
}

sub search_record {
    print "Enter Patient's name to search record: ";
    chomp(my $search_name = <STDIN>);

    open(my $fh, '<', 'MedicalDatabase.txt') or die "Could not open file 'MedicalDatabase.txt' $!";
    my $record_found = 0;
    while (my $line = <$fh>) {
        if ($line =~ /^$search_name$/i) {
            $record_found = 1;
        }
        if ($record_found) {
            print $line;
            if ($line =~ /^--------------$/) {
                $record_found = 0;
            }
        }
    }
    close($fh);

    print "Search completed.\n";
}

sub edit_record {
    print "Enter Patient's name to edit record: ";
    chomp(my $search_name = <STDIN>);
    system("perl -ni -e 'print unless /^$search_name/' MedicalDatabase.txt");
    add_record();
    print "Record edited successfully.\n";
}

while (1) {
    print "1. Add Patient Record\n";
    print "2. Delete Patient Record\n";
    print "3. Search Patient Record\n";
    print "4. Edit Patient Record\n";
    print "5. Exit Medlink\n";
    print "Enter your choice: ";
    chomp(my $choice = <STDIN>);

    if ($choice == 1) {
        add_record();
    } elsif ($choice == 2) {
        delete_record();
    } elsif ($choice == 3) {
        search_record();
    } elsif ($choice == 4) {
        edit_record();
    } elsif ($choice == 5) {
        exit;
    } else {
        print "Invalid choice\n";
    }
}
